### PR TITLE
Added SGX detection support

### PR DESCRIPTION
**Pull request by abourget**
Also digs in to see if the support is enabled or not.

### CODE REVIEW

File: cpuid.go [hunk 1]
```diff
@@ -747,6 +753,32 @@ func (c *CPUInfo) cacheSize() {
...
+		return
+	}
+
+	a, b, c, d := cpuidex(0x12, 0)
+	rval.SGX1Supported = a&0x01 != 0
+	rval.SGX2Supported = a&0x02 != 0
+	rval.MaxEnclaveSizeNot64 = int64(math.Pow(2, float64(d&0xFF)))
+	rval.MaxEnclaveSize64 = int64(math.Pow(2, float64((d>>8)&0xFF)))
+
+	fmt.Printf("e(abcd)x: %x %x %x %x\n", a, b, c, d)
```
Comment (klauspost) [inline cpuid.go]: I believe this is just a debug printf that isn't needed.

File: cpuid.go [hunk 2]
```diff
@@ -747,6 +753,32 @@ func (c *CPUInfo) cacheSize() {
...
+	rval.Available = available
+
+	if !available {
+		return
+	}
+
+	a, b, c, d := cpuidex(0x12, 0)
+	rval.SGX1Supported = a&0x01 != 0
+	rval.SGX2Supported = a&0x02 != 0
+	rval.MaxEnclaveSizeNot64 = int64(math.Pow(2, float64(d&0xFF)))
```
Comment (klauspost) [inline cpuid.go]: Instead of using float, I believe `1 << int64(d&0xFF)` will be 100% precise. Same for the following.

**Comment by klauspost** [comment]
Two minor things, otherwise it looks good.

If you can, please squash all changes into a single commit.

File: cpuid.go (hunk 2 above)
Comment (abourget) [inline cpuid.go]: good idea :)

**Comment by abourget** [comment]
There you go, tell me what you think :) thanks!

**Comment by abourget** [comment]
Are we good ?

**Comment by klauspost** [comment]
Looks great!  Sorry for the delay.

Thanks for the contribution!

**Pull request merged**